src/java/org/apache/sqoop/util/MainframeFTPClientUtils.java

```diff
@@ -207,8 +208,18 @@ public static FTPClient getFTPConnection(Configuration conf)
         throw new IOException("Could not login to server " + server
             + ":" + ftp.getReplyString());
       }
-      // set ASCII transfer mode
-      ftp.setFileType(FTP.ASCII_FILE_TYPE);
+      // set transfer mode
+      String transferMode = conf.get(MainframeConfiguration.MAINFRAME_FTP_TRANSFER_MODE);
```

Comment:
The whole getFTPConnection method is already very long and does a bunch of steps. You could consider refactoring it into smaller methods for readability.

Reply:
I'm not sure how you would refactor it. I can give it a try and see if it is what you're looking for.